frontend/taipy-gui/src/jsx/parser.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021-2025 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+import React from "react";
+
+const JSX_INTERPOLATION = /\{\!([a-zA-Z0-9_\.]+)\!\}/gs;
+
+const parseText = (txt: string, state?: Record<string, unknown>) => {
+    if (!JSX_INTERPOLATION.test(txt)) {
+        return txt;
+    } else {
+        return replaceInterpolations(txt, state);
+    }
+};
+
+const replaceInterpolations = (txt: string, state?: Record<string, unknown>) => {
+    JSX_INTERPOLATION.lastIndex = 0; // reset the regex state https://stackoverflow.com/questions/4724701/regexp-exec-returns-null-sporadically
+    let interpolations = JSX_INTERPOLATION.exec(txt);
```

Comment:
Does this recompile the regexp every time?
And I don't quite see how expressions are handled...